### ISSUE TITLE
crash on archlinux: error: Could not bind socket at ::1:29015 - Cannot assign requested address

### ISSUE THREAD

**Issue by weijar** [rethinkdb/rethinkdb#4000]
Hello,
I'm new to rethinkdb, I was installed rethinkdb on archlinux but can not to start it.

```
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: Running rethinkdb 1.16.3 (GCC 4.9.2)...
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: Running on Linux 3.14.37-1-lts x86_64
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: Loading data from directory /var/lib/rethinkdb/default
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: Version: rethinkdb 1.16.3 (GCC 4.9.2)
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: error: Error in src/arch/io/network.cc at line 793:
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: error: Could not bind socket at ::1:29015 - Cannot assign requested address
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: error: Backtrace:
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: error: Wed Apr  1 08:39:30 2015
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 1: backtrace_t::backtrace_t() at ??:?
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 2: format_backtrace(bool) at ??:?
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 3: report_fatal_error(char const*, int, char const*, ...) at ??:?
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 4: bind_ipv6_interface(int, int*, ip_address_t const&) at ??:?
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 5: linux_nonthrowing_tcp_listener_t::bind_sockets_internal(int*) at ??:?
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 6: linux_tcp_bound_socket_t::linux_tcp_bound_socket_t(std::set<ip_address_t, std::less<ip_address_t
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 7: connectivity_cluster_t::run_t::run_t(connectivity_cluster_t*, std::set<ip_address_t, std::less<i
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 8: do_serve(io_backender_t*, bool, base_path_t const&, metadata_persistence::cluster_persistent_fil
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 9: serve(io_backender_t*, base_path_t const&, metadata_persistence::cluster_persistent_file_t*, met
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 10: run_rethinkdb_serve(base_path_t const&, serve_info_t*, file_direct_io_mode_t, int, boost::optio
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 11: std::_Function_handler<void (), std::_Bind<void (*(base_path_t, serve_info_t*, file_direct_io_m
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 12: starter_t::run_wrapper(std::function<void ()> const&) at ??:?
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: 13: coro_t::run() at ??:?
Apr 01 08:39:30 wjzhou-pc rethinkdb[2211]: error: Exiting.
Apr 01 08:39:30 wjzhou-pc systemd-coredump[2237]: Coredump of 2211 (rethinkdb) is larger than configured processing limit, refusing.
Apr 01 08:39:30 wjzhou-pc systemd-coredump[2237]: Process 2211 (rethinkdb) of user 991 dumped core.
Apr 01 08:39:30 wjzhou-pc systemd[1]: rethinkdb@default.service: main process exited, code=killed, status=5/TRAP
Apr 01 08:39:30 wjzhou-pc systemd[1]: Unit rethinkdb@default.service entered failed state.
Apr 01 08:39:30 wjzhou-pc systemd[1]: rethinkdb@default.service failed.
```

**Comment by mglukhovsky** [comment]
@weijar: did you install RethinkDB by compiling from source, or by using [this Arch package](http://www.rethinkdb.com/docs/install/arch/)?

**Comment by weijar** [comment]
@mglukhovsky I am using the package from the arch community repository and using default config installed by that package (I was tried to bind ip to 127.0.0.1 but it can not resolve the problem).

**Comment by danielmewes** [comment]
Thank you for the bug report @weijar.
This seems to be similar to this old issue: https://github.com/rethinkdb/rethinkdb/issues/2405

You said that starting with `--bind 127.0.0.1` didn't resolve the issue. Did it result in the same error message or a different one?

Ping @Tryneus

**Comment by weijar** [comment]
@danielmewes 
I was tried
```
bind=127.0.0.1
```
and
```
port-offset=1
```
, both did result the same error message.

**Comment by Tryneus** [comment]
Looking into this.

**Comment by Tryneus** [comment]
@weijar, as a workaround (until we track this down and fix it), could you run with `--no-default-bind --bind 127.0.0.1`?

Got my arch VM back up and running with `1.16.x`, seeing if I can reproduce this.

**Comment by weijar** [comment]
@Tryneus It worked by add `--no-default-bind` option.

**Comment by Tryneus** [comment]
Could not reproduce this, but it was easy enough to fake, a fix is up in review 2781.

**Comment by weijar** [comment]
I ran rethinkdb's early version some months ago on archlinux's default kernel (3.18 or 3.19) and did not ran into this issue. For this crash I am running on linux-lts kernel (3.14).

**Comment by Tryneus** [comment]
The fix has been approved and merged to `next` in commit fb77595, and cherry-picked into `v2.0.x` in commit 2e512b6.  Will be in release 2.0.

Thanks for bringing this to our attention, @weijar, and I'm glad the workaround works for you in the meantime.
